### DISCUSSION TITLE
Dartls with mason

### DISCUSSION THREAD

**Discussion by StevenJPx2** [VonHeikemen/lsp-zero.nvim#40] [Q&A] [answered]
As of now, [@williambowman has decided to not not include language servers that already come with their corresponding executables in Mason.](https://github.com/williamboman/mason.nvim/issues/156#issuecomment-1198557932) This doesn't play well with `lsp-zero`, because not of the options `use, ensure_installed, configure` work with this. The only way I've got it to work is with having `lspconfig` setup the server separately. However in doing that I don't get any of the mappings out-of-the-box even with using `build_options`. What do I do?

**Comment by StevenJPx2** [comment, marked as answer]
Nvm this answered it! https://github.com/VonHeikemen/lsp-zero.nvim/discussions/28

By the way, @VonHeikemen, thank you for creating such an amazing out of the box experience!

**Reply by VonHeikemen** [reply]
I didn't know that about mason.nvim. That's kind of an issue now, I use mason's functions to check the installed servers.

Now I've added an option called `force_setup` to the function `.configure()`.

If you update lsp-zero this should work.

```lua
local lsp = require('lsp-zero')
lsp.preset('recommended')

lsp.configure('dartls', {force_setup = true})

lsp.setup()
```
